server/utils/uploadImage.js

```diff
@@ -0,0 +1,60 @@
+import multer from "multer";
+import multerS3 from "multer-s3";
+import { S3Client } from "@aws-sdk/client-s3";
+import path from "node:path";
+
+// create s3 instance using S3Client
+// (this is how we create s3 instance in v3)
+const s3 = new S3Client({
+	credentials: {
+		accessKeyId: process.env.ACCESS_KEY_ID, // store it in .env file to keep it safe
+		secretAccessKey: process.env.SECRET_ACCESS_KEY,
+	},
+	region: "eu-west-1", // this is the region that you select in AWS account
+});
+
+const s3Storage = multerS3({
+	s3: s3, // s3 instance
```

Comment:
Minor nitpick: If the variable name is the same, you can write it like this
```suggestion
	s3, // s3 instance
```